build-logic/jvm/src/main/kotlin/build-logic.java.gradle.kts

```diff
@@ -0,0 +1,48 @@
+import buildlogic.filterEolSimple
+
+plugins {
+    `java-base`
+    id("com.github.vlsi.gradle-extensions")
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_11
+    targetCompatibility = JavaVersion.VERSION_11
+}
+
+tasks.withType<JavaCompile>().configureEach {
+    inputs.property("java.version", System.getProperty("java.version"))
+    inputs.property("java.vm.version", System.getProperty("java.vm.version"))
+    options.apply {
+        encoding = "UTF-8"
+        compilerArgs.add("-Xlint:deprecation")
+        compilerArgs.add("-Werror")
+    }
+}
+
+tasks.withType<Javadoc>().configureEach {
+    (options as StandardJavadocDocletOptions).apply {
+        addBooleanOption("Xwerror", true)
+        addStringOption("sourcepath", "src/main/java")
+        // intentionally ignore missing errors for now
+        addBooleanOption("Xdoclint:all,-missing", true)
+    }
+}
+
+// Add default license/notice when missing (e.g. see :src:config that overrides LICENSE)
+
+tasks.withType<Jar>().configureEach {
+    into("META-INF") {
+        filterEolSimple("crlf")
+        from("$rootDir/LICENSE")
+        from("$rootDir/NOTICE")
+    }
+    manifest {
+        attributes["Bundle-License"] = "Apache-2.0"
+        attributes["Specification-Title"] = project.name + " " + project.description
+        attributes["Specification-Vendor"] = "dev.sigstore"
+        attributes["Implementation-Vendor"] = "dev.sigstore"
+        attributes["Implementation-Vendor-Id"] = "dev.sigstore"
+        // Implementation-Version is not here to make jar reproducible across versions
```

Comment:
does this really matter? Different versions of the jar don't necessarily need to be the same. Also I think we might need to include implementation-version to extract during runtime to set something like a user-agent string?

Reply:
It would be nice to have the same jar for both snapshot and non-snapshot build. For instance, I would like to have the same sha for 1.0.0-SNAPSHOT and 1.0.0 provided the code is the same. WDYT?

If you want to get the version in runtime, I incline to generating a simple class like SigstoreJavaVersion would be slightly better than storing version in the manifest.

I would expect that people would like to shade sigstore-java, and the manifest version is typically lost in that case, so manifest is not really reliable.

WDYT?

Reply:
Makes sense. Sgtm